### PR TITLE
Use parent RamAccounting instance in ShardDMLExecutor

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

The parent `RamAccounting` instance is either closed with a task or when
the final `RowConsumer` finishes.

Might fix flaky tests:

    KillIntegrationTest
    org.opentest4j.AssertionFailedError: [Query breaker not reset to 0 on node: node_s2]
      at __randomizedtesting.SeedInfo.seed([4C0486E0CAF108FE:C1CA1DD40B325C62]:0)
    expected: 0L
      but was: 644245L

As seen in https://wacklig.pipifein.dev/github/crate/crate/5146a935-7e26-4ae6-9bc6-cfb21f003d07/01b148f2-8c1c-4051-a686-bbb92ebe2763

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/execution/engine/indexing/ShardDMLExecutor.java [hunk 1]
```diff
@@ -220,10 +223,7 @@ public CompletableFuture<TResult> apply(BatchIterator<Row> batchIterator) {
             null,
             req -> nodeLimits.get(resolveNodeId(req)).getLastRtt(TimeUnit.MILLISECONDS)
         ).consumeIteratorAndExecute()
-            .thenApply(collector.finisher())
-            .whenComplete((result, err) -> {
```
Comment (matriv) [inline server/src/main/java/io/crate/execution/engine/indexing/ShardDMLExecutor.java:224]: We don't this now?

Comment (matriv) [reply server/src/main/java/io/crate/execution/engine/indexing/ShardDMLExecutor.java:224]: It's because the parent accounting is closed elsewhere, right?

Comment (mfussenegger) [reply server/src/main/java/io/crate/execution/engine/indexing/ShardDMLExecutor.java:224]: Yes, see commit message:

> The parent RamAccounting instance is either closed with a task or when
the final RowConsumer finishes.

**Review by BaurzhanSakhariev — APPROVED** [review]

**Review by matriv — APPROVED** [review]

**Comment by BaurzhanSakhariev** [comment]
Do we want this in 5.1?

**Comment by mfussenegger** [comment]
> Do we want this in 5.1?

Maybe after we have verified that it fixes the flakyness

**Pull request merged**